internal/provider/helpers/template.go

```diff
@@ -0,0 +1,28 @@
+package helpers
+
+import (
+	"bytes"
+	"text/template"
+)
+
+// RenderTemplate is a helper function to take a template and some configuration and
+// return the result as a string.
+//
+// Where:
+//   - tpl is a string like "Hello, {{.Name}}"
+//   - data is a struct like `mydata{Name: "World"}`
+func RenderTemplate(tpl string, data interface{}) string {
```

Comment:
The Terraform manifest building got really unwieldy with all of the positional arguments to `Sprintf`, so I used Go's `text/template` here so it's a bit easier to read and maintain.